src/download.rs

```diff
@@ -37,28 +38,30 @@ pub fn download_file(url: &str, file_path: &Path, redirect_path: &Path) -> Resul
 
   let status = response.status();
   if !status.is_success() {
-    fs::remove_file(redirect_path)?;
-    fs::remove_file(file_path)?;
     let err = read_error_response(response.text()?);
     anyhow::bail!(format!(
       "Failed to download from {}: {} {}",
       final_url, status, err
     ));
   }
 
-  let total_size = response
+  let content_len = response
     .headers()
     .get(reqwest::header::CONTENT_LENGTH)
     .and_then(|ct_len| ct_len.to_str().ok())
     .and_then(|ct_len| ct_len.parse::<u64>().ok())
-    .unwrap_or(0)
-    + file_size;
+    .unwrap_or(0);
+
+  let total_size = content_len + file_size;
 
   file.seek(SeekFrom::End(0))?;
-  let mut downloaded: u64 = file_size;
-  let mut last_reported_progress: i64 = -1;
+
+  const MEASUREMENT_SIZE: usize = 500;
+
+  let mut last_reported_progress: f64 = -1.0;
```

Comment:
NIT: consider using `Option<f64>` with `None` representing "not reported yet"
```suggestion
  let mut last_reported_progress: Option<f64> = None;
```